src/components/ProductListing/Filters.jsx

```diff
@@ -0,0 +1,221 @@
+                        productsDispatch({ 
+                          sortByType: btnType,
+                          filterType: "PRODUCTS_SORT_BY",
+                        })
+                      }
+                    />
+                    <i className="fa-solid fa-circle-check"></i>
+                    <p className="basic-chip-content">{name}</p>
+                  </label>
+                </li>
+              );
+            }
+          )}
+        </ul>
+        <ul className="checkbox-btn-container pb-10 outline-container p-5 b-radius-2 my-10">
+          <li className="no-list form-heading text-bold py-5 px-0">
+            Categories
+          </li>
+          {filtersData.categoryFilters.map(
+            ({ name, btnType }, index) => {
+              return (
+                <li className="no-list w-100 my-2" key={`category-${index}`}>
+                  <label className={`basic-chip flex-row align-center flex-wrap flex-gap-1 h6 filter-chip cursor-pointer ${productsState.categoryFilters[name] ? " fliter-chip-selected":""}`}>
+                    <input
+                      className="filters"
+                      type="checkbox"
+                      value={btnType}
+                      checked={
+                        productsState.categoryFilters[name] ? true : false
+                      }
+                      onChange={(e) =>
+                        productsDispatch({
+                          filterName: name,
+                          filterType: btnType,
+                          categoryFilters: ((categoryFilters, e) => {
+                            const newCategoryFilters = { ...categoryFilters };
+                            newCategoryFilters[name] = e.target.checked;
+                            return newCategoryFilters;
+                          })(productsState.categoryFilters, e)
+                        })
+                      }
+                    />
+                    <i className="fa-solid fa-circle-check"></i>
+                    <p className="basic-chip-content">{name}</p>
+                  </label>
+                </li>
+              );
+            }
+          )}
+        </ul>
+
+        <ul className="checkbox-btn-container pb-10 outline-container p-5 b-radius-2 my-10">
+          <li className="no-list form-heading text-bold py-5 px-0">Brand</li>
+          {filtersData.brandFilters.map(({ name, btnType }, index) => {
+            return (
+              <li className="no-list w-100 my-2" key={`brand-${index}`}>
+                <label className={`basic-chip flex-row align-center flex-wrap flex-gap-1 h6 filter-chip cursor-pointer ${productsState.brandFilters[name] ? " fliter-chip-selected":""}`}>
+                  <input
+                    className="filters"
+                    type="checkbox"
+                    value={btnType}
+                    checked={productsState.brandFilters[name] ? true : false}
+                    onChange={(e) =>
```

Comment:
One suggestion I have is that you could user changeHandlers and call the dispatch from there. This would separate your layout and logic.

Reply:
But again in that case I would have to create 2 to 3 handlers right @bharati-21 , or is there any way I can do that in one?